lib/dstate/interface.go

```diff
@@ -263,6 +263,12 @@ type ChannelState struct {
 	ThreadMetadata   *discordgo.ThreadMetadata `json:"thread_metadata"`
 
 	PermissionOverwrites []discordgo.PermissionOverwrite `json:"permission_overwrites"`
+
+	// The set of tags that can be used in a forum channel.
+	AvailableTags []discordgo.ForumTag `json:"available_tags"`
+
+	// The IDs of the set of tags that have been applied to a thread in a forum channel.
+	AppliedTags []discordgo.ForumTag `json:"applied_tags"`
```

Comment:
This part is little confusing, what is meant to be used here, slice of IDs or a slice of full ForumTag. Discord documentation states AppliedTags is an array of tag ID snowflakes not full objects

Reply:
Yeah discordgo is supposed to return []string with I think the snowflakes. I have the code manually upgrading them to full ForumTags by walking the AvailableTags slice. We could downgrade it to []int64 if that seems better

Reply:
I'd use IDs as int64 all around, even in AppliedTags slices - so here I'd use a helper function to convert from names users add to `complexThread "tags" (cslice ... )` as names to IDs.

But developer input needed, just my personal preference.